### PR TITLE
Testcases/resolved

### PR DESCRIPTION
**Pull request by Devyani606**
testcases for acceptance and controller

### CODE REVIEW

File: app/controllers/api/v1/admin/assessments_controller.rb [hunk 1]
```diff
@@ -9,13 +9,15 @@ class AssessmentsController < ApiController
 
         def index
           drives = Drive.where(is_assessment: true)
-          render json: { assessments: serialize_resource(drives, AssessmentSerializer) }
+          if drives
+            render_success(data: { assessments: serialize_resource(drives, AssessmentSerializer) },
```
Comment (anilmaurya) [inline app/controllers/api/v1/admin/assessments_controller.rb:13]: Please don't change format of API response, update test case if required

File: spec/acceptance/api/v1/admin/rules_controller_spec.rb [hunk 2]
```diff
@@ -74,22 +74,6 @@
           expect(status).to eq(200)
         end
       end
-
-      context 'with invalid params' do
```
Comment (anilmaurya) [inline spec/acceptance/api/v1/admin/rules_controller_spec.rb:78]: Do we need to delete test case or update it ?

File: spec/acceptance/api/v1/drives_candidate_spec.rb [hunk 3]
```diff
@@ -13,9 +13,9 @@
     create(:drive, updated_by_id: user.id, organization: organization,
                    created_by_id: user.id)
   end
-  let!(:drives_candidate) { create(:drives_candidate, candidate_id: candidate.id, drive_id: drive.id) }
-  let!(:drives_candidate1) { create(:drives_candidate, candidate_id: candidate1.id, drive_id: drive.id) }
-  let!(:drives_candidate2) { create(:drives_candidate, candidate_id: candidate2.id, drive_id: drive.id) }
+  let!(:drives_candidate) { create(:drives_candidate, candidate_id: candidate.id, drive_id: drive.id, drive_start_time: DateTime.current, drive_end_time: DateTime.current + 1.hours) }
```
Comment (anilmaurya) [inline spec/acceptance/api/v1/drives_candidate_spec.rb:16]: We can define it once and use for all
```
drive_start_time: DateTime.current, drive_end_time: DateTime.current + 1.hours
```

File: spec/acceptance/api/v1/execution_spec.rb [hunk 4]
```diff
@@ -8,9 +8,9 @@
     parameter :source_code, 'Source code'
     parameter :language_id, 'Language id'
     before do
-      stub_request(:post, 'http://roupi.xyz/submissions/?base64_encoded=false&wait=false')
+      stub_request(:post, 'https://ce.judge0.com/submissions/?base64_encoded=false&wait=false')
```
Comment (anilmaurya) [inline spec/acceptance/api/v1/execution_spec.rb:11]: Let's put  judge0 url in some constant and use at all places in acceptance specs

File: spec/controllers/api/v1/drives_candidate_spec.rb [hunk 5]
```diff
@@ -11,9 +11,9 @@
     @candidate2 = create(:candidate)
     @drive = create(:drive, updated_by_id: user.id, organization: organization,
                             created_by_id: user.id)
-    @drives_candidate = create(:drives_candidate, candidate_id: @candidate.id, drive_id: @drive.id)
-    @drives_candidate1 = create(:drives_candidate, candidate_id: @candidate1.id, drive_id: @drive.id)
-    @drives_candidate2 = create(:drives_candidate, candidate_id: @candidate2.id, drive_id: @drive.id)
+    @drives_candidate = create(:drives_candidate, candidate_id: @candidate.id, drive_id: @drive.id, drive_start_time: DateTime.current, drive_end_time: DateTime.current + 1.hours)
```
Comment (anilmaurya) [inline spec/controllers/api/v1/drives_candidate_spec.rb:14]: Try to remove duplicate code

File: spec/controllers/api/v1/results_controller_spec.rb [hunk 6]
```diff
@@ -8,27 +8,26 @@
...
       result = json
-      expect(result['data']['result'][0]['candidate_id']).to eq(@drives_candidate1.candidate_id)
-      expect(result['data']['result'][0]['first_name']).to eq('Kiran')
-      expect(result['data']['result'][0]['last_name']).to eq('Patil')
-      expect(result['data']['result'][0]['email']).to eq('kiran@gmail.com')
-      expect(result['data']['result'][0]['score']).to eq(8)
-      expect(result['data']['result'][0]['end_times']).to eq(@drives_candidate1.completed_at.iso8601.to_s)
-      expect(result['data']['result'][1]['candidate_id']).to eq(@drives_candidate2.candidate_id)
-      expect(result['data']['result'][1]['score']).to eq(10)
+      expect(result['data']['result'][1]['candidate_id']).to eq(@drives_candidate1.candidate_id)
```
Comment (anilmaurya) [inline spec/controllers/api/v1/results_controller_spec.rb:23]: Use some variable
var res = result['data']['result'][1]

for clean code

**Review by anilmaurya — COMMENTED** [review]
Please check review comment

File: spec/controllers/api/v1/results_controller_spec.rb (hunk 6 above)
Comment (Devyani606) [reply spec/controllers/api/v1/results_controller_spec.rb:23]: done

File: app/controllers/api/v1/admin/assessments_controller.rb (hunk 1 above)
Comment (Devyani606) [reply app/controllers/api/v1/admin/assessments_controller.rb:13]: done

File: spec/acceptance/api/v1/admin/rules_controller_spec.rb (hunk 2 above)
Comment (Devyani606) [reply spec/acceptance/api/v1/admin/rules_controller_spec.rb:78]: done

File: spec/acceptance/api/v1/drives_candidate_spec.rb (hunk 3 above)
Comment (Devyani606) [reply spec/acceptance/api/v1/drives_candidate_spec.rb:16]: it is required for multiple submission

File: spec/controllers/api/v1/drives_candidate_spec.rb (hunk 5 above)
Comment (Devyani606) [reply spec/controllers/api/v1/drives_candidate_spec.rb:14]: required for multiple submission

File: spec/acceptance/api/v1/execution_spec.rb (hunk 4 above)
Comment (Devyani606) [reply spec/acceptance/api/v1/execution_spec.rb:11]: done

File: spec/acceptance/api/v1/drives_candidate_spec.rb (hunk 3 above)
Comment (Devyani606) [reply spec/acceptance/api/v1/drives_candidate_spec.rb]: done

File: spec/controllers/api/v1/drives_candidate_spec.rb (hunk 5 above)
Comment (Devyani606) [reply spec/controllers/api/v1/drives_candidate_spec.rb]: done

**Comment by anilmaurya** [comment]
@Devyani606 Please do not use `rubocop:disable all`, we should only disable abc complexity whenever required

**Pull request merged**